### ISSUE TITLE
Parsing -----|  takes a long time

### ISSUE THREAD

**Issue by julesxoxzo** [Python-Markdown/markdown#1095]
Hi,

Am having problems with parsing this particular string in markdown:
`----------------------------------------------------------------------------------------------------|` [1]

The `convert` takes too long to parse that particular string that it ends up using all compute time of our on-premise dev server. 

I tried the following strings to test:
- `--------------------------`
- `|--------------------------`
- `|--------------------------|`
- `|----------------------------------------------------------------------------------------------------|`
- `|----------------------------------------------------------------------------------------------------`

All of the above worked except the one I mentioned ([1]). 

I could only make it work by doing any of the ff:
-  Enclose in back ticks: 
```
`----------------------------------------------------------------------------------------------------|` - works
```
- Enclose in triple back ticks: 
```
     ```
      ----------------------------------------------------------------------------------------------------| - also works. 
     ```
```

Any help will be appreciated.

**Comment by facelessuser** [comment]
I'm confused, I have no problem with this:

```
>>> import markdown
>>> markdown.markdown('----------------------------------------------------------------------------------------------------| [1]')
'<p>----------------------------------------------------------------------------------------------------| [1]</p>'
```

Can you provide a minimal example with relevant extensions you are using?

**Comment by julesxoxzo** [comment]
Hi,

Sure. here.

here's a minimal example:

```
from markdown import Markdown

string = "----------------------------------------------------------------------------------------------------|"

md = Markdown()
converted = md.convert(string)
print(converted)
```

Here's the one with extensions I'm using.
```
from mdx_linkify.mdx_linkify import LinkifyExtension
from markdown import Markdown

def add_rel_norefferer(attrs, new=False):
    attrs[(None, "rel")] = "noreferrer"

    return attrs

string = "----------------------------------------------------------------------------------------------------|"

md = Markdown(
    extensions=[
        LinkifyExtension(linker_options={"callbacks": [add_rel_norefferer]}),
        "fenced_code",
        "nl2br",
    ],
)
converted = md.convert(string)
print(converted)
```

Both take a long time to run.

- current version of markdown I'm using is `Markdown==3.3.2`

**Comment by julesxoxzo** [comment]
Hi @facelessuser , 

I updated my package to `3.3.3` and it now works! Sorry for the hassle and thanks for the help!

**Comment by waylan** [comment]
This is a duplicate of #1055 which was fixed in #1056 and was included in version 3.3.3.
